langstream-runtime/langstream-runtime-impl/src/main/java/ai/langstream/runtime/agent/AgentRunner.java

```diff
@@ -1036,6 +1035,15 @@ public BadRecordHandler getBadRecordHandler() {
         public TopicConnectionProvider getTopicConnectionProvider() {
             return topicConnectionProvider;
         }
+
+        @Override
+        public void criticalFailure(Throwable error) {
+            log.error(
+                    "Critical failure: %s. Shutting down the runtime..."
+                            .formatted(error.getMessage()),
+                    error);
+            AgentContext.super.criticalFailure(error);
```

Comment:
We can make the behaviour configurable with a static "Consumer<Throwable>" variable, and in AbstractApplicationRunner (that is used in unit tests) we intercept the call without crashing the JVM